### PR TITLE
PLFM-6986: Add datasets entry to managerMap

### PR DESCRIPTION
**Pull request by nickgros**
Adding this to the configuration prevents error here when calling certain table services on a dataset: https://github.com/Sage-Bionetworks/Synapse-Repository-Services/blob/780f716dacdec2fc7cec6741bcfe1f87b2e00733/services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableTransactionManagerProviderImpl.java#L29

### CODE REVIEW

File: services/repository-managers/src/main/resources/private/managers-spb.xml [hunk 1]
```diff
@@ -393,6 +393,9 @@
 				<entry key="submissionview">
 					<ref bean="tableViewTransactionManager" />
 				</entry>
+				<entry key="dataset">
+					<ref bean="tableViewTransactionManager" />
+				</entry>
```
Comment (nickgros) [inline services/repository-managers/src/main/resources/private/managers-spb.xml]: Move configuration to code and add integration test to DatasetIntegrationTest

File: services/repository-managers/src/main/resources/private/managers-spb.xml [hunk 2]
```diff
@@ -382,19 +382,6 @@
...
-				</entry>
-				<entry key="entityview">
-					<ref bean="tableViewTransactionManager" />
-				</entry>
-				<entry key="submissionview">
-					<ref bean="tableViewTransactionManager" />
-				</entry>
-			</map>
-		</property>
 	</bean>
```
Comment (nickgros) [inline services/repository-managers/src/main/resources/private/managers-spb.xml]: Wasn't sure if I should remove the whole bean, so I only removed the map. I think replacing all of the XML in this file with annotations is a different task.

Comment (john-hill) [reply services/repository-managers/src/main/resources/private/managers-spb.xml]: I agree that the rest of the file is beyond this task.  However, we should move tableTransactionManagerProvider out of the xml.

**Review by john-hill — APPROVED** [review]
Looks good!  Thanks.

**Pull request merged**